dbt/adapters/snowflake.py

```
@@ -0,0 +1,244 @@
+
+    date_function = 'CURRENT_TIMESTAMP()'
+
+    @classmethod
+    def acquire_connection(cls, profile):
+
+        # profile requires some marshalling right now because it includes a
+        # wee bit of global config.
+        # TODO remove this
+        credentials = copy.deepcopy(profile)
+
+        credentials.pop('type', None)
+        credentials.pop('threads', None)
+
+        result = {
+            'type': 'snowflake',
+            'state': 'init',
+            'handle': None,
+            'credentials': credentials
+        }
+
+        logger.debug('Acquiring snowflake connection')
+
+        if flags.STRICT_MODE:
+            validate_connection(result)
+
+        return cls.open_connection(result)
+
+    @staticmethod
+    def hash_profile(profile):
+        return ("{}--{}--{}--{}--{}".format(
+            profile.get('account'),
+            profile.get('database'),
+            profile.get('schema'),
+            profile.get('user'),
+            profile.get('warehouse'),
+        ))
+
+    @classmethod
+    def open_connection(cls, connection):
+        if connection.get('state') == 'open':
+            logger.debug('Connection is already open, skipping open.')
+            return connection
+
+        result = connection.copy()
+
+        try:
+            credentials = connection.get('credentials', {})
+            handle = snowflake.connector.connect(
+                account=credentials.get('account'),
+                user=credentials.get('user'),
+                password=credentials.get('password'),
+                database=credentials.get('database'),
+                schema=credentials.get('schema'),
+                warehouse=credentials.get('warehouse'),
+                autocommit=False
+            )
+
+            result['handle'] = handle
+            result['state'] = 'open'
+        except snowflake.connector.errors.Error as e:
+            logger.debug("Got an error when attempting to open a snowflake "
```

does this fail silently?